src/scene/gsplat/gsplat-data.js

```diff
@@ -408,6 +412,77 @@ class GSplatData {
             })
         }], false);
     }
+
+    calcMortonOrder() {
+        // https://fgiesen.wordpress.com/2009/12/13/decoding-morton-codes/
+        const encodeMorton3 = (x, y, z) => {
+            const Part1By2 = (x) => {
+                x &= 0x000003ff;
+                x = (x ^ (x << 16)) & 0xff0000ff;
+                x = (x ^ (x <<  8)) & 0x0300f00f;
+                x = (x ^ (x <<  4)) & 0x030c30c3;
+                x = (x ^ (x <<  2)) & 0x09249249;
+                return x;
+            };
+
+            return (Part1By2(z) << 2) + (Part1By2(y) << 1) + Part1By2(x);
+        };
+
+        const x = this.getProp('x');
+        const y = this.getProp('y');
+        const z = this.getProp('z');
+
+        const minX = x.reduce((a, b) => Math.min(a, b), x[0]);
+        const minY = y.reduce((a, b) => Math.min(a, b), y[0]);
+        const minZ = z.reduce((a, b) => Math.min(a, b), z[0]);
+
+        const maxX = x.reduce((a, b) => Math.max(a, b), x[0]);
+        const maxY = y.reduce((a, b) => Math.max(a, b), y[0]);
+        const maxZ = z.reduce((a, b) => Math.max(a, b), z[0]);
```

Comment:
This looks lovely and tidy, but I do wonder whether something like this would be significantly faster:

```
function getMinMax(arr) {
    let min = arr[0];
    let max = arr[0];
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] < min) min = arr[i];
        if (arr[i] > max) max = arr[i];
    }
    return { min, max };
}

const { min: minX, max: maxX } = getMinMax(x);
const { min: minY, max: maxY } = getMinMax(y);
const { min: minZ, max: maxZ } = getMinMax(z);
```

Reply:
Yep thanks, I've made the change.